Sources/ComposableArchitecture/Observation/NavigationStack+Observation.swift

```diff
@@ -502,11 +566,12 @@ extension StackState {
     }
 
     public static func == (lhs: Self, rhs: Self) -> Bool {
-      lhs.id == rhs.id
+      lhs.id == rhs.id && lhs.element.id == rhs.element.id
```

Comment:
- `id`: index in Stack
  - To distinguish the screen navigation from
- `element.id`: `Path.State.id`
  - To distinguish the `NavigationLink` in the same screen with the same destination state type